python/pyspark/sql/types.py

```diff
@@ -212,15 +213,29 @@ def needConversion(self) -> bool:
 
     def toInternal(self, dt: datetime.datetime) -> int:
         if dt is not None:
-            seconds = (
-                calendar.timegm(dt.utctimetuple()) if dt.tzinfo else time.mktime(dt.timetuple())
-            )
+            if platform.system().lower() == 'windows':
```

Comment:
Should we also check if the value is negative?

Reply:
Don't need, this method is still available when the negative.

Reply:
How about performance? This is hot path that's executed every value, and the performance here is pretty critical.

Reply:
Are there professional performance testers?